frontend/src/page/Realm.tsx

```diff
@@ -27,8 +56,48 @@ export const Realm: React.FC<Props> = ({ match }) => {
     </React.Fragment>;
 };
 
-const DUMMY_NAMES: { [key: string]: string } = {
-    "lectures": "Lectures",
-    "math": "Department of Mathematics",
-    "algebra": "Algebra I",
+// Looks up each segment of the `/` separated `path` as realm and returns a list
+// of realm ids starting with 0 (root).
+const resolvePath = (path: string): number[] | null => {
+    const ids = [0];
+    for (const segment of path == "" ? [] : path.split("/")) {
```

Comment:
```suggestion
    for (const segment of path === "" ? [] : path.split("/")) {
```

But actually, I would probably early-return before the loop, `if (!path)`. **Or** just ignore that case; I mean does this even come up? And what about all the other edge cases like `"/"` or `"foobar"`? :thinking:

Reply:
Yes it does come up for the index page. There, the path is empty. But I thought a couple of times about having a separate `page/Home.tsx` for that. I think we should do it eventually, as there are increasingly more differences between home page and realm pages.

Reply:
Makes sense.